Sources/Apodini/SyntaxTreeVisitor/DelegatingHandlerInitializerVisitor.swift

```diff
@@ -0,0 +1,126 @@
+//
+// Created by Andreas Bauer on 02.07.21.
+//
+
+import Foundation
+import OrderedCollections
+import ApodiniUtils
+
+class DelegatingHandlerInitializerVisitor: HandlerVisitor {
+    let semanticModelBuilder: SemanticModelBuilder?
+    unowned let visitor: SyntaxTreeVisitor
+
+
+    /// The index of the next element in the ContextKey value which we haven't
+    /// considered yet. This is required as the `initializer` property will most certainly
+    /// remove some elements from the original context key value.
+    private var nextContextKeyIndex: Int = 0
+
+    private var initializers: OrderedSet<DelegatingHandlerContextKey.Entry> = []
+
+    /// Defines if `visit` is called for the first/main Handler
+    private var firstHandler = true
+
+    init(calling builder: SemanticModelBuilder?, with visitor: SyntaxTreeVisitor) {
+        self.semanticModelBuilder = builder
+        self.visitor = visitor
+
+        self.queryInitializers()
+    }
+
+    func visit<H: Handler>(handler: H) throws {
+        preconditionTypeIsStruct(H.self, messagePrefix: "Delegating Handler")
+
+        // we only look at the Delegates for the "main" Handler (also to not get into an infinite loop)
+        // The other ones instantiated via DelegatingHandlerInitializer are covered below.
```

Comment:
Where "below" exactly? I really can't find where this would be handled. Note that a `DelegatingHandlerInitializer` might also construct a type that manually stacks `Delegate`s outside of its given `Handler` (i.e. the one you already analyzed).

Reply:
Not sure what the comment points to either 😅 
The PR currently ignores Delegates on Delegating Handlers instantiated via an initializer, as its a bit tricky to not fall into the case where we create a loop parsing the Handler passed to the `.instance` method. We could try to do a type based approach for that, to filter those out, but not sure how stable that would be.

Reply:
The latest commits adds support for parsing Metadata defined on Delegates declared inside Handlers instantiated via `DelegatingHandlerInitializer`.
To avoid the infinite loop, I relied on a comparing of the Type against the last parsed Handler.